master/buildbot/util/service.py

```diff
@@ -264,6 +268,7 @@ def reconfigServiceWithBuildbotConfig(self, new_config):
                                     **config_sibling._config_kwargs)
 
     def setServiceParent(self, parent):
+        # XXX(sa2ajj): can we make 'master' a simple read-only property instead?
```

Comment:
I guess we should not try to optimize what has *not* shown to be a bottleneck. I'm not convinced a property will be so much a bottleneck.
I agree with sa2ajj that a property would be a good option for the problem of the master reference.

```
import time
class A(object):
   @property
   def master(self):
      return self.parent.parent

a = A()
b = A()
c = A()
c.parent = b
b.parent = a
t1 = time.time()
for i in xrange(10000000):
    d = c.master
t2 = time.time()
for i in xrange(10000000):
    d = c.parent
t3 = time.time()
print t2-t1, t3-t2
% python test.py
3.43862700462 1.08624792099
```
ok, this property is 3.4 times slower than direct access, but you can still make more than 2M per second...